plugins/woocommerce/includes/wc-formatting-functions.php

```diff
@@ -1293,7 +1293,7 @@ function wc_format_weight( $weight ) {
 	$weight_string = wc_format_localized_decimal( $weight );
 
 	if ( ! empty( $weight_string ) ) {
-		$weight_string .= ' ' . get_option( 'woocommerce_weight_unit' );
+		$weight_string .= ' ' . __(get_option( 'woocommerce_weight_unit' ), 'woocommerce');
```

Comment:
I understand your intention here, but this is generally not a great way to translate things. In fact, if you run this through the linter, it's flagged as an error:

`WordPress.WP.I18n.NonSingularStringLiteralText`

The main reason for this is that various translation tools will be unable to successfully parse this.

Reply:
> Thank you for submitting this—however I think it may be worth considering a different approach.
> 
> Interestingly enough, we do already translate units such as 'kg' (and you can see this working within the **WooCommerce ▸ Settings ▸ Products ▸ General** admin screen). However, it is also true that they are not currently translated on the frontend (such as within the _Additional Information_ area on single product posts), which I think is the problem you are addressing.
> 
> For consistency, it is probably best to translate once and then use those translations everywhere else. We do essentially this very thing with strings such as country names (take a look in the `i18n/` directory to see what I mean), and it would probably make sense to apply the same strategy here.

Thanks for the answer. Any approach will suit me, the main thing is that the problem should be solved. I'm not a WP developer, so I believe, you know better what to do.

Reply:
Yes, I agree this ought to be improved: these units should be translatable and the translations ought to be used on the frontend and anywhere else that is appropriate.

If you or @masteradhoc wish to make changes per these notes, please feel free (and let me know if I can answer any other questions). Alternatively, #31263 is already labelled as an enhancement and so the core team will ultimately tackle it if you don't want to—though, that may take a little longer as there are other priorities ahead of this one.